docs/src/content/docs/book/exit-codes.mdx

```diff
@@ -3,323 +3,671 @@ title: Exit codes
+repeat (256) {
+    nativeReserve(ton("0.001"), ReserveAtMost);
 }
 ```
 
-### $133$: Contract stopped {#133}
-
-The stoppable trait allows to stop the contract.
-If you send a message to a stopped contract, and the contract asks for a check by running ```self.requireNotStopped();```, this exit code will be thrown.
-In the current version of Tact, 40368 exit code will be thrown instead of 133.
+### 34: Invalid or unsupported action {#34}
 
-Example:
+There are only four supported actions at the moment: changing the contract code, sending a message, reserving a specific amount of [nanoToncoins](/book/integers#nanotoncoin) and changing the library cell. If there's any issue with the specified action (invalid message, unsupported action, etc.), an error with exit code $34$ is thrown: `Invalid or unsupported action`.
 
 ```tact
-import "@stdlib/deploy";
-import "@stdlib/ownable";
-import "@stdlib/stoppable";
+// For example, let's try to send an ill-formed message:
+nativeSendMessage(emptyCell(), 0); // won't fail in compute phase,
+                                   // but will result in exit code 34 in Action phase
+```
 
-message FakeLaunch {}
+### 35: Invalid source address in outbound message {#35}
 
-contract Fireworks with
-    Deployable,
-    Ownable,
-    Stoppable,
-{
-    owner: Address;
-    stopped: Bool;
+If the source address in the outbound message isn't equal to [`addr_none`](https://docs.ton.org/develop/data-formats/msg-tlb#addr_none00) or to the address of the contract that initiated this message, an error with exit code $35$ is thrown: `Invalid source address in outbound message`.
 
-    init() {
-        self.owner = sender();
-        self.stopped = false;
-    }
+### 36: Invalid destination address in outbound message {#36}
 
-    receive(msg: FakeLaunch) {
-        self.stopped = true;
-        self.requireNotStopped();
-    }
-}
+If the destination address in the outbound message is invalid, e.g. it doesn't conform to the relevant [TL-B][tlb] schemas, contains unknown workchain ID or it has invalid length for the given workchain, an error with exit code $36$ is thrown: `Invalid destination address in outbound message`.
+
+:::note
+
+  If the [optional flag +2](/book/message-mode#optional-flags) is set, this error won't be thrown and the given message won't be sent.
+
+:::
+
+### 37: Not enough Toncoin {#37}
+
+If all the inbound message value with [base mode 64](/book/message-mode#base-modes) set has been already consumed and there's not enough funds to pay for the failed action, or the [TL-B][tlb] layout of the provided value ([`CurrencyCollection`](https://docs.ton.org/develop/data-formats/msg-tlb#currencycollection)) is invalid, or there's not enough funds to pay [forward fees](https://docs.ton.org/develop/smart-contracts/guidelines/processing) or not enough funds after deducting fees, an error with exit code $37$ is thrown: `Not enough Toncoin`.
```

Comment:
"If all the inbound"   ====> "If every inbound message"?

Reply:
Not quite, but I'll correct it